src/appleDeviceUtils/appleDeviceUtils.js

```diff
@@ -0,0 +1,109 @@
+// Do not remove -- Apple devices change this when they support modern webkit messaging
+let hasModernWebkitAPI = false
+// INJECT hasModernWebkitAPI HERE
+
+// The native layer will inject a randomised secret here and use it to verify the origin
+const secret = 'PLACEHOLDER_SECRET'
+
+const ddgGlobals = window.navigator.ddgGlobals
+
+/**
+ * Sends message to the webkit layer (fire and forget)
+ * @param {String} handler
+ * @param {*} data
+ * @returns {*}
+ */
+const wkSend = (handler, data = {}) =>
+    window.webkit.messageHandlers[handler].postMessage(data)
+
+/**
+ * Generate a random method name and adds it to the global scope
+ * The native layer will use this method to send the response
+ * @param {String} randomMethodName
+ * @param {Function} callback
+ */
+const generateRandomMethod = (randomMethodName, callback) => {
+    Object.defineProperty(ddgGlobals.window, randomMethodName, {
+        enumerable: false,
+        // configurable, To allow for deletion later
+        configurable: true,
+        writable: false,
+        // Use proxy to ensure stringification isn't possible
+        value: new Proxy(function () {}, {
+            apply (target, thisArg, args) {
+                callback(...args)
+                delete ddgGlobals.window[randomMethodName]
+            }
+        })
+    })
+}
+
+/**
+ * Sends message to the webkit layer and waits for the specified response
+ * @param {String} handler
+ * @param {*} data
+ * @returns {Promise<*>}
+ */
+const wkSendAndWait = async (handler, data = {}) => {
+    if (hasModernWebkitAPI) {
+        return wkSend(handler, data)
+    }
+
+    // Older versions
```

Comment:
What does this comment mean?

Reply:
A leftover of the initial pseudocode. Nuked it now, thanks.